pycaret/internal/preprocess/preprocessor.py

```diff
@@ -843,14 +843,39 @@ def _balance(self, fix_imbalance_method):
             smotetomek=SMOTETomek,
         )
 
+        METHODS_WITH_RANDOM_STATE_SUPPORT = [
+            "CondensedNearestNeighbour"
+            "InstanceHardnessThreshold"
+            "OneSidedSelection"
+            "RandomUnderSampler"
+            "SMOTE"
+            "SMOTENC"
+            "SMOTEN"
+            "ADASYN"
+            "BorderlineSMOTE"
+            "KMeansSMOTE"
+            "SVMSMOTE"
+            "SMOTEENN"
+            "SMOTETomek"
+        ]
+
         if isinstance(fix_imbalance_method, str):
             fix_imbalance_method = fix_imbalance_method.lower()
             if fix_imbalance_method not in strategies:
                 raise ValueError(
                     "Invalid value for the strategy parameter, got "
                     f"{fix_imbalance_method}. Choose from: {', '.join(strategies)}."
                 )
-            balance_estimator = FixImbalancer(strategies[fix_imbalance_method]())
+            if (
+                strategies[fix_imbalance_method].__class__
+                in METHODS_WITH_RANDOM_STATE_SUPPORT
+                and session_id is not None
+            ):
+                balance_estimator = FixImbalancer(
+                    strategies[fix_imbalance_method](random_state=session_id)
+                )
+            else:
+                balance_estimator = FixImbalancer(strategies[fix_imbalance_method]())
```

Comment:
Could we somehow use duck typing (try to use `random_state` as an argument inside a try-except block, with fallback to no `random_state`) instead of hardcoding a list of supported classes? That way it will be much easier to maintain. Other alternative would be to use `inspect` to check if `random_state` is in the init signature, but duck typing will probably be easier.

Reply:
Something like eg.
```
try:
	balance_estimator = FixImbalancer(
	    strategies[fix_imbalance_method](random_state=session_id)
	)
except TypeError:
	balance_estimator = FixImbalancer(
	    strategies[fix_imbalance_method]()
	)
```

Reply:
Done